jsonnet/dashboards/apps/loki.libsonnet

```diff
@@ -0,0 +1,246 @@
+local grafana = import 'grafonnet/grafana.libsonnet';
+local dashboard = grafana.dashboard;
+local prometheus = grafana.prometheus;
+local graphPanel = grafana.graphPanel;
+local row = grafana.row;
+local statPanel = grafana.statPanel;
+local msgs_graph(title,target) =
+  graphPanel.new(
+          title=title,
+          datasource='$datasource',
+          format='bytes',
+          stack=false,
+          nullPointMode='null as zero',
+          legend_current=true,
+        ).addTarget(prometheus.target(expr="histogram_quantile(0.95, sum(rate("+target+")) by (le,route))",legendFormat='{{route}}'));
+local ingester_graph(title,target,legend_format, format='short')= graphPanel.new(
+          title=title,
+          datasource='$datasource',
+          format=format,
+          stack=false,
+          nullPointMode='null as zero',
+          legend_current=true,
+          legend_values=true,
+          fill=0,
+          linewidth=1,
+          legend_sort='current'
+        ).addTarget(prometheus.target('histogram_quantile(0.95, sum(rate('+target+')) by (le))',legendFormat=legend_format));
+local index_graph(title,target,legend_format, format='short')= graphPanel.new(
+          title=title,
+          datasource='$datasource',
+          format=format,
+          stack=false,
+          nullPointMode='null as zero',
+          legend_current=true,
+          legend_values=true,
+          fill=0,
+          linewidth=1,
+          legend_sort='current'
+        ).addTarget(prometheus.target('histogram_quantile(0.95, sum(rate('+target+')) by (le))',legendFormat=legend_format));
+local querier_graph(title,target,legend_format, format='short')= graphPanel.new(
+          title=title,
+          datasource='$datasource',
+          format=format,
+          stack=false,
+          nullPointMode='null as zero',
+          legend_current=true,
+          legend_values=true,
+          fill=0,
+          linewidth=1,
+          legend_sort='current'
+        ).addTarget(prometheus.target(target,legendFormat=legend_format));
+
+{
+  grafanaDashboards+:: {
+    loki:
+    local version =
+        statPanel.new(
+          title = "Loki Version",
+          datasource = "$datasource",
+          fields = "/^version$/",
+          reducerFunction ="last",
+          graphMode = 'none',
+          colorMode = 'background',
+        )
+        .addTarget(prometheus.target(format='table',expr="loki_build_info"));
+    local msgs =
+        statPanel.new(
+          title='Log Messages Total',
+          datasource='$datasource',
+          graphMode='none',
+        )
+        .addTarget(
+          prometheus.target('sum(log_messages_total)')
```

Comment:
Metric with name `log_messages_total` can have a lot of k8s metrics exporters, you need to use `job` to specify component. And it is nice to not hardcode it but to use grafana templates - [cadvisor example](https://github.com/dNationCloud/kubernetes-monitoring/blob/main/jsonnet/dashboards/apps/cadvisor.libsonnet#L36)